### ISSUE TITLE
Missing images in email notifications? (needs more research)

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#9927]
With the new Google Storage paperclip fixes, I wasn't 100% sure images were getting properly sent in email notifications. But this may not be related to storage so much as our templates... we'll have to compare to other recent email notifications. But for reference, here is a comment that includes an image:

https://publiclab.org/notes/warren/12-12-2010/front-illumination-ccds-used-digital-cameras-dont-perform-well-uv#c29037

![image](https://user-images.githubusercontent.com/24359/126370372-ea38406c-1759-4fcb-8a7b-b348a4929b40.png)

I got the below email and it doesn't include an image tag.

```
Subject: New comment on Front-illumination CCDs (as used in digital cameras) don't perform well in UV (#28) - #c29037
Mime-Version: 1.0
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: 7bit
Authentication-Results: spool.mail.gandi.net; dkim=pass header.d=publiclab-org.20150623.gappssmtp.com header.s=20150623 header.b=EyPE+qq8; dmarc=none; spf=pass (spool.mail.gandi.net: domain of do-not-reply@publiclab.org designates 209.85.219.46 as permitted sender) smtp.mailfrom=do-not-reply@publiclab.org

Hi! There's been a response to your research note '<a href='https://publiclab.org/notes/warren/12-12-2010/front-illumination-ccds-used-digital-cameras-dont-perform-well-uv'>Front-illumination CCDs (as used in digital cameras) don&#39;t perform well in UV</a>'.
You can reply to this email or visit this link:

<p>https://publiclab.org/notes/warren/12-12-2010/front-illumination-ccds-used-digital-cameras-dont-perform-well-uv#c29037</p>

<p><b><a href="https://publiclab.org/profile/icarito">icarito</a> wrote:</b></p>

<hr /> 

<p>
</p>

<hr />

<p>Look like spam?
    Mark this as <a href="https://publiclab.org/admin/mark_comment_spam/29037">Spam</a>
</p>


<div style="color:#aaa;">

<p>Reply at: https://publiclab.org/notes/warren/12-12-2010/front-illumination-ccds-used-digital-cameras-dont-perform-well-uv#c29037</p>

<p>Report abuse to: moderators@publiclab.org</p>

Check out the blog at https://publiclab.org/blog | Love our work? Make a one-time or recurring donation at https://publiclab.org/donate



If this email title has an ID in the format #0000, you can reply with the email you use at PublicLab.org and your response will be posted as a comment on the website.

</div>
```

**Comment by jywarren** [comment]
Here's the template:

https://github.com/publiclab/plots2/blob/02315eb4a298949c8cd9ece772121b645787ddfa/app/views/comment_mailer/notify_note_author.html.erb#L18

Let's check the server logs to see if it was in the outgoing email. 

Then we can manually run `auto_link(RDiscount.new(@comment.body_email(ActionMailer::Base.default_url_options[:host])).to_html)` to confirm that.

**Comment by jywarren** [comment]
OK, indeed - `auto_link()` is stripping the images out:

```ruby
[11] pry(main)> include ActionView::Helpers
=> Object
[12] pry(main)> auto_link(RDiscount.new(comment.body_email(ActionMailer::Base.default_url_options[:host])).to_html)
=> "hi  there\n"
[13] pry(main)> (RDiscount.new(comment.body_email(ActionMailer::Base.default_url_options[:host])).to_html)=> "<p>hi <img src='https://i.publiclab.org/system/images/photos/000/039/681/medium/smd.png'> there</p>\n"
```

**Comment by jywarren** [comment]
I don't think it's related but just in case: https://github.com/publiclab/plots2/issues/2354

auto_link auto-sanitizes, which may be the issue: https://github.com/tenderlove/rails_autolink/

If we don't use the built-in sanitizer we should use the customized one here:

https://github.com/publiclab/plots2/blob/8daad1a70d022810b249a3d7882f6c4bd4fe3727/app/models/comment.rb#L513-L520

However we're relying on this shared method, which is not well-linked to the sanitizer:

https://github.com/publiclab/plots2/blob/8daad1a70d022810b249a3d7882f6c4bd4fe3727/app/models/concerns/comments_shared.rb#L7-L15

**Comment by jywarren** [comment]
I believe i have a fix in #9929
